pkg/sources/reconciler/kafkasource/adapter.go

```diff
@@ -0,0 +1,256 @@
+/*
+Copyright 2022 TriggerMesh Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kafkasource
+
+import (
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+
+	"knative.dev/eventing/pkg/reconciler/source"
+	"knative.dev/pkg/apis"
+
+	commonv1alpha1 "github.com/triggermesh/triggermesh/pkg/apis/common/v1alpha1"
+	"github.com/triggermesh/triggermesh/pkg/apis/sources/v1alpha1"
+	common "github.com/triggermesh/triggermesh/pkg/reconciler"
+	"github.com/triggermesh/triggermesh/pkg/reconciler/resource"
+)
+
+const (
+	envBootstrapServers      = "BOOTSTRAP_SERVERS"
+	envTopics                = "TOPICS"
+	envGroupID               = "GROUP_ID"
+	envUsername              = "USERNAME"
+	envPassword              = "PASSWORD"
+	envSecurityMechanisms    = "SECURITY_MECHANISMS"
+	envSSLCA                 = "SSL_CA"
+	envSSLClientCert         = "SSL_CLIENT_CERT"
+	envSSLClientKey          = "SSL_CLIENT_KEY"
+	envSSLInsecureSkipVerify = "SSL_INSECURE_SKIP_VERIFY"
+
+	envSalsEnable = "SALS_ENABLE"
+	envTLSEnable  = "TLS_ENABLE"
+
+	envKerberosConfigPath  = "KERBEROS_CONFIG_PATH"
+	envKerberosKeytabPath  = "KERBEROS_KEYTAB_PATH"
+	envKerberosServiceName = "KERBEROS_SERVICE_NAME"
+	envKerberosRealm       = "KERBEROS_REALM"
+	envKerberosUsername    = "KERBEROS_USERNAME"
+	envKerberosPassword    = "KERBEROS_PASSWORD"
+
+	krb5ConfPath   = "/etc/krb5.conf"
+	krb5KeytabPath = "/etc/krb5.keytab"
+)
+
+// adapterConfig contains properties used to configure the target's adapter.
+// Public fields are automatically populated by envconfig.
+type adapterConfig struct {
+	// Configuration accessor for logging/metrics/tracing
+	configs source.ConfigAccessor
+	// Container image
+	Image string `default:"gcr.io/triggermesh/kafkasource-adapter"`
+}
+
+// Verify that Reconciler implements common.AdapterBuilder.
+var _ common.AdapterBuilder[*appsv1.Deployment] = (*Reconciler)(nil)
+
+// BuildAdapter implements common.AdapterBuilder.
+func (r *Reconciler) BuildAdapter(src commonv1alpha1.Reconcilable, sinkURI *apis.URL) (*appsv1.Deployment, error) {
+	typedSrc := src.(*v1alpha1.KafkaSource)
+
+	var secretVolumes []corev1.Volume
+	var secretVolMounts []corev1.VolumeMount
+
+	if typedSrc.Spec.KerberosAuth != nil {
```

Comment:
There is a method `MaybeAppendValueFromEnvVar()` that exsists in the `"github.com/triggermesh/triggermesh/pkg/reconciler"` common package, that looks cleaner imo.  Not asking for changes :) this is fine. 

```
authEnvs = common.MaybeAppendValueFromEnvVar(authEnvs, common.EnvAADClientSecret, spAuth.ClientSecret)
```